airflow/api_connexion/endpoints/dataset_endpoint.py

```diff
@@ -124,3 +130,153 @@ def get_dataset_events(
     return dataset_event_collection_schema.dump(
         DatasetEventCollection(dataset_events=events, total_entries=total_entries)
     )
+
+
+def _generate_queue_event_where_clause(
+    *,
+    dag_id: str | None = None,
+    dataset_id: int | None = None,
+    uri: str | None = None,
+    before: str | None = None,
+) -> list:
+    """Get DatasetDagRunQueue where clause."""
+    where_clause = []
+    if dag_id is not None:
+        where_clause.append(DatasetDagRunQueue.target_dag_id == dag_id)
+    if dataset_id is not None:
+        where_clause.append(DatasetDagRunQueue.dataset_id == dataset_id)
+    if uri is not None:
+        where_clause.append(DatasetModel.uri == uri)
+    if before is not None:
+        where_clause.append(DatasetDagRunQueue.created_at < format_datetime(before))
+    return where_clause
+
+
+@security.requires_access_dataset("GET")
+@provide_session
+def get_dag_dataset_queue_event(
+    *, dag_id: str, uri: str, before: str | None = None, session: Session = NEW_SESSION
+) -> APIResponse:
+    """Get a queued Dataset event for a DAG."""
+    where_clause = _generate_queue_event_where_clause(dag_id=dag_id, uri=uri, before=before)
+    ddrq = session.scalar(
+        select(DatasetDagRunQueue)
+        .join(DatasetModel, DatasetDagRunQueue.dataset_id == DatasetModel.id)
+        .where(*where_clause)
+    )
+    if ddrq is None:
+        raise NotFound(
+            "Queue event not found",
+            detail=f"Queue event with dag_id: `{dag_id}` and dataset uri: `{uri}` was not found",
+        )
+    queue_event = {"created_at": ddrq.created_at, "dag_id": dag_id, "uri": uri}
+    return queue_event_schema.dump(queue_event)
+
+
+@security.requires_access_dataset("DELETE")
```

Comment:
```suggestion
@security.requires_access_dataset("DELETE")
@security.requires_access_dag("GET")
```

Though read is probably enough to check when talking delete?

Reply:
Yep, I think read is enough in this case. Just added it. Thanks!